### PR TITLE
b/WPC-655: As a publisher, i want mailto-Links without slashes (//)

### PR DESCRIPTION
**Pull request by elias-summermatter**

### CODE REVIEW

File: packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx [hunk 1]
```diff
@@ -137,6 +147,14 @@ export function LinkMenu() {
   )
 }
 
+async function validateMail(mail: string) {
+  if (mail) {
+    const pattern = new RegExp('^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}$', 'i')
```
[the remark this answers is not in the document]
Comment (elias-summermatter) [reply packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx:152]: Yes but the one from the rfc looks like this and i taught this is a bit to much for an optional validation and no one understands this one the one is a simplified one that is easy readable which covers most of the cases:
 `(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])`

File: packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx [hunk 2]
```diff
@@ -104,7 +109,12 @@ export function LinkMenu() {
 
             <FormControl value={url} onChange={url => setURL(url)} />
           </InputGroup>
-          <HelpBlock>{url && !isValidURL ? t('blocks.richText.invalidLink') : undefined}</HelpBlock>
+          {prefix !== prefixType.mailto && url && !isValidURL ? (
```
[the remark this answers is not in the document]
Comment (elias-summermatter) [reply packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx:112]: I taught about that but for me its more "react" stile like this and its simpler if one oft the other reviewers disagrees with me on this one i can change it.

**Review by penk85 — APPROVED** [review]

**Review by camilleory — APPROVED** [review]

File: packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomaszdurka) [reply packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx:112]: I am not a fan of using if not-mailto for all the other urls.

File: packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomaszdurka) [reply packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx:152]: Is there any other concern than readibility? If no I would go for RFC and just store under constant.

File: packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx [hunk 3]
```diff
@@ -36,12 +36,17 @@ export function LinkMenu() {
...
   const [isValidURL, setIsValidURL] = useState(false)
+  const [isValidMail, setIsValidMail] = useState(false)
   const isDisabled = !url || !title
 
   const {t} = useTranslation()
 
   useEffect(() => {
-    validateURL(url).then(value => setIsValidURL(value))
+    if (prefix === prefixType.mailto) {
+      validateMail(url).then(value => setIsValidMail(value))
```
Comment (tomaszdurka) [inline packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx:46]: Guys, let's rather implement `validateLink(prefix, url)` and only use `isValidLink`.

**Review by tomaszdurka — CHANGES_REQUESTED** [review]

**Pull request merged**